src/main/webapp/filter_tag_area.js

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+class FilterTagArea {
+ 
+  constructor(searchArea) {
+    this.parentSearchArea = searchArea;
+
+    /* Active filter area holds the add filter button, the new filter text box, and any active filters */
+    this.activeFilterArea = document.createElement("div");
+    this.activeFilterArea.setAttribute("class", "filter-holder");
+
+    /* filterEntryArea is displayed when the user clicks add filter. It first asks for a type of filter, 
+       then asks the user to enter the keyword for that filter type */
+    this.filterEntryArea = document.createElement("div");
+    this.filterEntryArea.setAttribute("class", "filter-tag-area");
+    this.filterEntryArea.setAttribute("id", "filter-entry");
+
+    this.filterEntryClose = document.createElement("div");
```

Comment:
You're gonna run into a merge conflict because i rearranged the order a little bit and changed some styling but you can just keep everything in this pr and I can redo it after you merge